neo4jgraph.py

```diff
@@ -0,0 +1,457 @@
+            Dataset collection id, is unique.
+
+        Raises
+        ------
+        ServiceUnavailable
+            Web socket error that occurs when having problems connecting to neo4j.
+        """
+        with self.driver.session() as session:
+            try:
+                generated_id = session.write_transaction(self._generate_dataset_collection, dataset_collection_id)
+                logging.info(f"Generated dataset collection {generated_id}.")
+
+            except ServiceUnavailable as exception:
+                logging.error(exception)
+                raise
+
+    @staticmethod
+    def _generate_dataset_collection(tx, dataset_collection_id):
+        """Creates and runs a query for dataset collection creation."""
+        query = (
+            f'MERGE (dataset_collection:dataset_collection {{dataset_collection_id: {dataset_collection_id}, '
+            f'description: "Dataset collection number {dataset_collection_id}"}}) '
+            f'RETURN dataset_collection'
+        )
+
+        query_output = tx.run(query)  # Returns an iterative with one dataset collection
+        logging.info(f"Run cypher query to generate dataset collection: {query}.")
+        return [row["dataset_collection"]["dataset_collection_id"] for row in query_output][0]
+
+    def generate_system_collection(self, system_collection_id):
+        """Generates a system collection node with id and description based on the id.
+        Creating a system collection with id 1 will create a node with these attributes:
+            system_collection_id: 1,
+            description: System collection number 1.
+
+        Parameters
+        ----------
+        system_collection_id : int
+            System collection id, is unique.
+
+        Raises
+        ------
+        ServiceUnavailable
+            Web socket error that occurs when having problems connecting to neo4j.
+        """
+        with self.driver.session() as session:
+            try:
+                generated_id = session.write_transaction(self._generate_system_collection, system_collection_id)
+                logging.info(f"Generated system collection {generated_id}")
+
+            except ServiceUnavailable as exception:
+                logging.error(exception)
+                raise
+
+    @staticmethod
+    def _generate_system_collection(tx, system_collection_id):
+        """Creates and runs a query for system collection creation."""
+        query = (
+            f'MERGE (system_collection:system_collection {{system_collection_id: {system_collection_id}, '
+            f'description: "System collection number {system_collection_id}"}}) '
+            f'RETURN system_collection'
+        )
+
+        query_output = tx.run(query)  # Returns an iterative with one system collection
+        logging.info(f"Run cypher query to generate system collection: {query}.")
+        return [row["system_collection"]["system_collection_id"] for row in query_output][0]
```

Comment:
Isn't this equivalent to query_output[0]["system_collection"]["system_collection_id"]? If so this one seems simpler. Same elsewhere

Reply:
This won't work, as neo4j result doesn't support indexing, but it's iterable, so changed it to next(iter(query_output))["dataset_collection"]["dataset_collection_id"].